docs/user-guide/src/bmo/reboot_annotation.md

```diff
@@ -0,0 +1,69 @@
+# Reboot annotation
+
+As the title implies, there is an annotation for rebooting a BareMetalHost.
+The annotation key takes either of the following forms:
+
+- `reboot.metal3.io`
+- `reboot.metal3.io/{key}`
+
+In its basic form (`reboot.metal3.io`), the annotation will trigger a reboot of the BareMetalHost.
+The controller will remove the annotation as soon as it has restored power to the host.
+
+The advanced form (`reboot.metal3.io/{key}`) includes a unique suffix (indicated with `{key}`).
+In this form the host will be kept in `PoweredOff` state until the annotation has been removed.
+This can be useful if some tasks needs to be performed while the host is in a known stable state.
+The purpose of the `{key}` is to allow multiple clients to use the API simultaneously in a safe way.
+Each client chooses a key and touches only the annotations that has this key to avoid interfering with other clients.
+
+If there are multiple annotations, the controller will wait for all of them to be removed (by the clients) before powering on the host.
+Similarly, if both forms of annotations are used, the `reboot.metal3.io/{key}` form will take precedence.
+This ensures that the host stays powered off until all clients are ready (i.e. all annotations are removed).
+
+The annotation value should be a JSON map containing the key `mode` and a value `hard` or `soft` to indicate if a hard or soft reboot should be performed.
+It is not necessary to specify the annotation value.
+In case it is omitted, the default is to first try a soft reboot, and if that fails, do a hard reboot.
+
+The exact behavior of `hard` and `soft` reboot depends on the Ironic configuration.
+Please see the [Ironic configuration reference](https://docs.openstack.org/ironic/latest/configuration/config.html) for more details on this.
```

Comment:
Hmm maybe I need to change it a bit. The link has a list of Ironic settings that will impact how the hard and soft reboot works. But of course the Ironic documentation does not mention anything about BMO, so it can be a bit hard to understand which settings will have an impact and in what way.

The point was to show that there are settings in Ironic, like `soft_power_off_timeout` for example that are relevant to this feature. Any suggestion on how to make it clearer?

Reply:
Got it, actually it maybe tricky because those ironic settings needs to be configured in Metal3 ironic configuration to tune it, i.e `power_state_change_timeout` is the [one](https://github.com/metal3-io/ironic-image/blob/main/ironic-config/ironic.conf.j2#L85)  which is also mentioned in the link you mentioned. But `soft_power_off_timeout` is not set for example and maybe it is not good idea to claim they are related? Also, have not seen any reference to ironic docs in the original proposal document.

Reply:
I don't really get the question, some of these options are not set in our ironic config because we use deffault values and in some cases some of these were overlooked for one reason or an other some were implemented/changed during the development life cycle of metal3 . 

So I would do it in a way that let's highlight the ones that we are using in our ironic image and just put there a link that points to the other additional options as a reference.

Reply:
Ok, how about this: I add an example to make it clear that in this list of variables there are some that are related to reboot.

```suggestion
Please see the [Ironic configuration reference](https://docs.openstack.org/ironic/latest/configuration/config.html) for more details on this, e.g. the `soft_power_off_timeout` variable is relevant.
```

Reply:
I have edited my answer a bit, so I think you are free to include the soft and the hard but I am also satisfied with this solution also to just provide an example.

Reply:
Done